src/main/scala/org/broadinstitute/dsde/workbench/avram/util/ErrorResponse.scala

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import java.util.logging.Logger
+
+import com.google.api.server.spi.response._
+
+/**
+  * Representation of an error response for use throughout the code. Is eventually translated to one
+  * of the Google Cloud Endpoints Framework exceptions that, when thrown, results in the correct
+  * HTTP response.
+  *
+  * @param statusCode HTTP status code
+  * @param message    error message
+  */
+case class ErrorResponse(statusCode: Int, message: String) {
+  private val log = Logger.getLogger(getClass.getName)
+
+  /**
+    * Create a Google Cloud Endpoints Framework exception matching the status code.
+    */
+  def exception: Exception = {
+    log.info(message)
+    statusCode match {
+      case 400 => new BadRequestException(message)
```

Comment:
Might make a `StatusCode` type to avoid having to remember the numbers, and maybe reduce the code here.

Reply:
+1